### PR TITLE
fix: Fix flame package example app

### PR DESCRIPTION
**Pull request by luanpotter**
# Description

Our main example for the flame package was no longer working due to some changes on the input system, namely the need to manually mark events as handled.
I also removed some pausing logic that I do not think belongs on the main example, that can be a separate topic on the dashbook example. This example should be simple and contain key learning topics.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples`.

## Breaking Change

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/flame/example/lib/main.dart [hunk 1]
```diff
@@ -80,6 +67,7 @@ class Square extends PositionComponent with Tappable {
   @override
   bool onTapUp(TapUpInfo info) {
     removeFromParent();
+    info.handled = true;
```
Comment (spydon) [inline packages/flame/example/lib/main.dart:70]: Why was this needes?

**Review by spydon — APPROVED** [review]

**Review by renancaraujo — APPROVED** [review]

File: packages/flame/example/lib/main.dart (hunk 1 above)
Comment (luanpotter) [reply packages/flame/example/lib/main.dart:70]: because returning true no longer sets it automatically

Comment (spydon) [reply packages/flame/example/lib/main.dart:70]: But why does it need to be set?

Comment (luanpotter) [reply packages/flame/example/lib/main.dart:70]: so the check to avoid adding a new square works

Comment (spydon) [reply packages/flame/example/lib/main.dart:70]: Ah, but just return false instead to stop the propagation, and no check is needed.

File: packages/flame/example/lib/main.dart [hunk 2]
```diff
@@ -80,6 +67,7 @@ class Square extends PositionComponent with Tappable {
...
     return true;
```
Comment (spydon) [inline packages/flame/example/lib/main.dart:71]: Here 👆

File: packages/flame/example/lib/main.dart (hunk 1 above)
Comment (luanpotter) [reply packages/flame/example/lib/main.dart:70]: let me try

Comment (luanpotter) [reply packages/flame/example/lib/main.dart:70]: I thought `true` meant handled

Comment (luanpotter) [reply packages/flame/example/lib/main.dart:70]: doesn't work

Comment (spydon) [reply packages/flame/example/lib/main.dart:70]: `true` means continue to propagate, but it always ends up on the game it seems like? I wonder if that is a bug...
You can check `info.continuePropagation` on the game, but then it's cleaner to do lile you did before.

Comment (spydon) [reply packages/flame/example/lib/main.dart:70]: Yeah, it must have always been like that, since that is the entry point, I'll merge this

**Pull request merged**